maap/Result.py

```diff
@@ -23,71 +22,65 @@ def getLocalPath(self, destpath=".", overwrite=False):
         url = self._location
         destfile = self._downloadname.replace('/', '')
 
-        # Downloadable url does not exist
         if not url:
+            # Downloadable url does not exist
             return None
         if url.startswith('ftp'):
             if not overwrite and not os.path.isfile(destpath + "/" + destfile):
                 urllib.urlretrieve(url, destpath + "/" + destfile)
 
             return destpath + '/' + destfile
         elif url.startswith('s3'):
-            o = urlparse(url)
-            filename = url[url.rfind("/") + 1:]
-            if not overwrite and not os.path.isfile(destpath + "/" + filename):
-                s3 = boto3.client('s3', aws_access_key_id=self._awsKey, aws_secret_access_key=self._awsSecret)
-                s3.download_file(o.netloc, o.path.lstrip('/'), destpath + "/" + filename)
+            try:
+                o = urlparse(url)
+                filename = url[url.rfind("/") + 1:]
+                if not overwrite and not os.path.isfile(destpath + "/" + filename):
+                    s3 = boto3.client('s3')
+                    s3.download_file(o.netloc, o.path.lstrip('/'), destpath + "/" + filename)
+            except:
+                # Fall back to HTTP
+                http_url = self._convertS3toHttp(url)
+                return self._getHttpData(http_url, overwrite, destpath, destfile)
```

Comment:
I guess we are making and can make an assumption that all HTTP(S) links in the MAAP metadata store will be using this federated token at this point since we are curating the metadata. I imagine that there could be case with different metadata that is not using EDL federated login over HTTPS or an S3 link but it shouldn't be the case now that I can think of. Perhaps we could add a comment to that effect in the code that we are assuming there are not other cases than:
1. s3://nasa-maap-data-store
2. https which will work for federated login using the right header

Are those 2 cases your understanding as well Brian in having the getData() method have this conditional layout?